errors/errctx.go

```diff
@@ -48,3 +49,60 @@ func Wrapf(ctx context.Context, err error, format string, args ...interface{}) e
 func Errorf(ctx context.Context, format string, args ...interface{}) error {
 	return ErrCtx{ctx: ctx, err: errors.Errorf(format, args...)}
 }
+
+// GetDeepestContext unwrap all wrapped errors from err to get the deepest context
+// from ErrCtx errors. If there is no wrapped ErrCtx, GetDeepestContext returns nil.
+func GetDeepestContext(err error) context.Context {
```

Comment:
Small question about the naming, why not using `RootCtx` to match the other method names?